src/go/configgenerator/route_generator.go

```diff
@@ -248,15 +249,15 @@ func makeHttpRouteMatcher(httpRule *commonpb.Pattern) *routepb.RouteMatch {
 			},
 		}
 	} else {
+		if err := validateRegexProgramSize(regex, util.GoogleRE2MaxProgramSize); err != nil {
+			return nil, fmt.Errorf("invalid route path regex: %v", err)
```

Comment:
error is not clear.  The error message should
"invalid regex {} from UriTemplate {}"